src/battle-tooltips.ts

```diff
@@ -0,0 +1,1514 @@
+				value.itemModify(item.naturalGift.basePower);
+			}
+		}
+		// Moves which have base power changed according to weight
+		if (['lowkick', 'grassknot', 'heavyslam', 'heatcrash'].includes(move.id)) {
+			let isGKLK = ['lowkick', 'grassknot'].includes(move.id);
+			if (target) {
+				let targetWeight = target.getWeightKg();
+				let pokemonWeight = pokemon.getWeightKg(serverPokemon);
+				let basePower;
+				if (isGKLK) {
+					basePower = 20;
+					if (targetWeight >= 200) basePower = 120;
+					else if (targetWeight >= 100) basePower = 100;
+					else if (targetWeight >= 50) basePower = 80;
+					else if (targetWeight >= 25) basePower = 60;
+					else if (targetWeight >= 10) basePower = 40;
+				} else {
+					basePower = 40;
+					if (pokemonWeight > targetWeight * 5) basePower = 120;
+					else if (pokemonWeight > targetWeight * 4) basePower = 100;
+					else if (pokemonWeight > targetWeight * 3) basePower = 80;
+					else if (pokemonWeight > targetWeight * 2) basePower = 60;
+				}
+				value.set(basePower);
+			} else {
+				value.setRange(isGKLK ? 20 : 40, 120);
+			}
+		}
+		if (!value.value) return value;
+
+		// Other ability boosts
+		if (pokemon.status === 'brn' && move.category === 'Special') {
+			value.abilityModify(1.5, "Flare Boost");
+		}
+		if (move.flags['pulse']) {
+			value.abilityModify(1.5, "Mega Launcher");
+		}
+		if (move.flags['bite']) {
+			value.abilityModify(1.5, "Strong Jaw");
+		}
+		if (value.value <= 60) {
+			value.abilityModify(1.5, "Technician");
+		}
+		if (['psn', 'tox'].includes(pokemon.status) && move.category === 'Physical') {
+			value.abilityModify(1.5, "Toxic Boost");
+		}
+		if (['Rock', 'Ground', 'Steel'].includes(moveType) && this.battle.weather === 'sandstorm') {
+			if (value.tryAbility("Sand Force")) value.weatherModify(1.3, "Sandstorm", "Sand Force");
+		}
+		if (move.secondaries) {
+			value.abilityModify(1.3, "Sheer Force");
+		}
+		if (move.flags['contact']) {
+			value.abilityModify(1.3, "Tough Claws");
+		}
+		if (target) {
+			if (["MF", "FM"].includes(pokemon.gender + target.gender)) {
+				value.abilityModify(1.25, "Rivalry");
+			} else if (["MM", "FF"].includes(pokemon.gender + target.gender)) {
+				value.abilityModify(0.75, "Rivalry");
+			}
+		}
+		const noTypeOverride = ['judgment', 'multiattack', 'naturalgift', 'revelationdance', 'struggle', 'technoblast', 'weatherball'];
+		if (move.type === 'Normal' && move.category !== 'Status' && !noTypeOverride.includes(move.id)) {
+			value.abilityModify(this.battle.gen > 6 ? 1.2 : 1.3, "Aerilate");
+			value.abilityModify(this.battle.gen > 6 ? 1.2 : 1.3, "Galvanize");
+			value.abilityModify(this.battle.gen > 6 ? 1.2 : 1.3, "Pixilate");
+			value.abilityModify(this.battle.gen > 6 ? 1.2 : 1.3, "Refrigerate");
+		}
+		if (move.flags['punch']) {
+			value.abilityModify(1.2, 'Iron Fist');
+		}
+		if (move.recoil || move.hasCustomRecoil) {
+			value.abilityModify(1.2, 'Reckless');
+		}
+		if (this.battle.gen > 6) {
```

Comment:
I believe it was established that Normalize doesn’t boost moves that change type on their own, so this should be grouped with the -ates.

Reply:
Er, it should at least check `noTypeOverride`

Reply:
For the record, this is totally not in the part of the code that I changed...